src/pip/_internal/utils/setuptools_build.py

```diff
@@ -64,7 +67,10 @@ def make_setuptools_shim_args(
     args = [sys.executable]
     if unbuffered_output:
         args += ["-u"]
-    args += ["-c", _SETUPTOOLS_SHIM.format(setup_py_path)]
+    if sys.version_info[:2] >= (3, 12):
```

Comment:
```suggestion
    if sys.version_info >= (3, 12):
```

Don’t think you need to slice!